### PR TITLE
Add google maps

### PR DESCRIPTION
**Pull request by RazvanPop**
Add google maps and 2 markers

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -9,11 +9,16 @@
     "lint": "next lint"
   },
   "dependencies": {
+    "@googlemaps/js-api-loader": "^1.13.8",
+    "@react-google-maps/api": "^2.7.0",
+    "@types/google-maps-react": "^2.0.5",
```
Comment (florea-razvan) [inline package.json:14]: please remove this dependency and add it as a dev-dependency

File: pages/GoogleMap.tsx [hunk 2]
```diff
@@ -0,0 +1,103 @@
+import React, { useEffect, useState } from 'react'
+import { GoogleMap, InfoWindow, Marker, useJsApiLoader } from '@react-google-maps/api';
+import { eventNames } from 'process';
+
+
+interface IMarker{
+  address:string;
```
Comment (florea-razvan) [inline pages/GoogleMap.tsx:7]: please use indentation in between field name and field type

File: pages/GoogleMap.tsx [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+
+const marker2 ={
+  lat: 46.7546919, 
+  lng: 23.5604183
+  
+}
+
+
+
+// const marker3 = new google.maps.LatLng(46.7545548,23.558355);
```
Comment (florea-razvan) [inline pages/GoogleMap.tsx:34]: remove commented code

File: pages/GoogleMap.tsx [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+
+  const [map, setMap] = useState<GoogleMap>()
+
+  const onLoad = React.useCallback(function callback(map) {
+    const bounds = new window.google.maps.LatLngBounds();
+    map.fitBounds(bounds);
+    setMap(map)
+  }, [])
+
+  // const onUnmount = React.useCallback(function callback(map)  {
```
Comment (florea-razvan) [inline pages/GoogleMap.tsx:54]: commented code

**Review by florea-razvan — CHANGES_REQUESTED** [review]

File: pages/GoogleMap.tsx [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+    const bounds = new window.google.maps.LatLngBounds();
+    map.fitBounds(bounds);
+    setMap(map);
+  }, []);
+
+  const initEventListener = (): void => {
+    if (map) {
+      google.maps.event.addListener(map, "click", function (e: any) {
+        coordinateToAddress(e.latLng);
+      });
```
Comment (vasileorza) [inline pages/GoogleMap.tsx:52]: Please favor using arrow functions instead of functions, whenever possible

File: pages/GoogleMap.tsx [hunk 6]
```diff
@@ -0,0 +1,111 @@
...
+  const onLoad = React.useCallback(function callback(map) {
+    const bounds = new window.google.maps.LatLngBounds();
+    map.fitBounds(bounds);
+    setMap(map);
+  }, []);
+
+  const initEventListener = (): void => {
+    if (map) {
+      google.maps.event.addListener(map, "click", function (e: any) {
+        coordinateToAddress(e.latLng);
```
Comment (vasileorza) [inline pages/GoogleMap.tsx:51]: Please move the contents of the method directly here, avoid defining methods outside react hooks.

File: pages/GoogleMap.tsx [hunk 7]
```diff
@@ -0,0 +1,111 @@
...
+  const coordinateToAddress = async (coordinate: GoogleLatLng) => {
+    const geocoder = new google.maps.Geocoder();
+    geocoder.geocode({ location: coordinate }, function (results, status) {
+      if (status === "OK") {
+        if (results) {
+          setMarker({
+            address: results[0].formatted_address,
+            latitude: coordinate.lat(),
+            longitude: coordinate.lng(),
+          });
```
Comment (vasileorza) [inline pages/GoogleMap.tsx:66]: Please remove the `marker` state, and copy the add marker directly here.

**Review by vasileorza — APPROVED** [review]

**Pull request merged**